controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -96,6 +96,18 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 
 	expectedSnapshot, err := a.prepareSnapshotForPipelineRun(a.pipelineRun, a.component, a.application)
 	if err != nil {
+		// If PipelineRun result returns cusomized error stop reconciliation and update PLR annotation
+		if h.IsMissingInfoInPipelineRunError(err) {
+			// update the build PLR annotation with the error cusomized Reason and Value
+			err = a.annotateBuildPipelineRunWithCreateSnapshotAnnotation(err)
```

Comment:
can you use different variable name than `err`? On line 106 reusing this variable name causes logging wrong error

Reply:
nice catch , 
changed